frontend/public/components/factory/list-page.jsx

```diff
@@ -197,6 +197,15 @@ export const FireMan_ = connect(null, {filterList})(
         textFilter,
       } = this.props;
 
+      const runOrNavigate = itemName => {
+        const action = createProps.action ? createProps.action(itemName) : null;
+        if (action) {
+          action();
```

Comment:
nit: better to scope `action` to the `if` block since that's the only place it's used

```js
if (createProps.action) {
  const action = actionProps.action(itemName);
  action();
} else if (createProps.createLink) {
  history.push(createProps.createLink(itemName));
}
```

Reply:
With recent change, the `action()` is called twice. Not a big deal as the implementation is expected to be simple `switch` statement, but it is called twice...
The `action` property can be `undefined` or return a false-value for the given parameter. In both cases, the `createLink` is used as the default.

I am fine even with the recent change as there is no real performance impact, just wanted to point it out.